source/common/network/apple_dns_impl.cc

```diff
@@ -0,0 +1,259 @@
+#include "common/network/apple_dns_impl.h"
+
+#include <dns_sd.h>
+
+#include <chrono>
+#include <cstdint>
+#include <list>
+#include <memory>
+#include <string>
+
+#include "envoy/common/platform.h"
+
+#include "common/common/assert.h"
+#include "common/common/fmt.h"
+#include "common/network/address_impl.h"
+#include "common/network/utility.h"
+
+#include "absl/strings/str_join.h"
+
+namespace Envoy {
+namespace Network {
+
+AppleDnsResolverImpl::AppleDnsResolverImpl(Event::Dispatcher& dispatcher)
+    : dispatcher_(dispatcher) {
+  ENVOY_LOG(warn, "Constructing DNS resolver");
```

Comment:
note to clean up logging in this file before final review.

Reply:
Yep, was planning on doing when I move from draft.